src/components/tabs/__internal__/tabs-header/tabs-header.style.ts

```diff
@@ -165,27 +99,89 @@ const StyledVerticalTabsWrapper = styled.div<StyledVerticalTabsWrapperProps>`
   flex-direction: column;
 `;
 
-const StyledTabsBottomBorderWrapper = styled.div`
+type StyledWrapperProps = {
+  align: string;
+  position: string;
+  role?: string;
+  extendedLine?: boolean;
+  noRightBorder?: boolean;
+  isInSidebar?: boolean;
+};
+
+type StyledNavigationButtonWrapperProps = {
+  position: string;
+  visible: boolean;
+};
+
+const StyledWrapper = styled.section<StyledWrapperProps>`
+  overflow: hidden;
+  position: relative;
+  display: flex;
+  ${({ extendedLine = true }) =>
+    !extendedLine &&
+    css`
+      width: fit-content;
+    `}
+
+  ${({ position }) => position === "top" && "white-space: nowrap"};
+
+  ${({ align }) =>
+    align === "right" &&
+    css`
+      justify-content: flex-end;
+      text-align: right;
+    `}
+`;
+
+const StyledNavigationButtonWrapper = styled.div<StyledNavigationButtonWrapperProps>`
+  width: 48px;
+  height: 48px;
+
+  ${({ position, visible }) => css`
+    display: ${visible ? "block" : "none"};
+    ${position === "right" && "right: 0; position: absolute;"}
+  `}
+`;
+
+const StyledNavigationButton = styled.button`
```

Comment:
**issue**: I noticed some visual glitches when a tab title, partially covered by a navigation button, seems to overlap the button when focused or if it has a validation border.

Here's an example when a covered tab title is focused: 

https://github.com/user-attachments/assets/44cdd721-8d71-468e-b6f2-2658f1545939

And another when a covered tab title has a validation border:

![navigation-button-validation-border](https://github.com/user-attachments/assets/ed002c46-b638-459c-ba34-c416c82c2d43)


To avoid this glitch, perhaps we could adjust the scrolling behaviour to ensure that a partially covered tab title is fully in view when focused. As for the issue with validation borders, I guess we could either explore increasing the z-index of navigation buttons, or adjusting the scroll offset so tab titles are never partially visible.

What do you think?